app/src/main/java/uk/ac/ebi/atlas/search/metadata/JsonMultiexperimentCellTypeMarkerGenesController.java

```diff
@@ -28,17 +31,26 @@ public JsonMultiexperimentCellTypeMarkerGenesController(
     }
 
     @GetMapping(value = "/json/cell-type-marker-genes/{cellType}",
-                produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
     public String getCellTypeMarkerGenes(
             @PathVariable String cellType,
             @RequestParam(name = "experiment-accessions", required = false) Collection<String> experimentAccessions) {
         return GSON.toJson(
                 highchartsHeatmapAdapter.getMarkerGeneHeatmapDataSortedLexicographically(
                         experimentAccessions == null ?
                                 multiexperimentCellTypeMarkerGenesService.getCellTypeMarkerGeneProfile(
-                                        URLDecoder.decode(cellType, StandardCharsets.UTF_8)) :
+                                        getDecodedCellType(cellType)) :
                                 multiexperimentCellTypeMarkerGenesService.getCellTypeMarkerGeneProfile(
                                         ImmutableSet.copyOf(experimentAccessions),
-                                        URLDecoder.decode(cellType, StandardCharsets.UTF_8))));
+                                        getDecodedCellType(cellType))));
+    }
+
+    private static @NotNull String getDecodedCellType(@NotNull String cellType) {
```

Comment:
Do we need the @NotNull annotations here? I know that IntelliJ is adding that to your code when you probably extracted out from the `getCellTypeMarkerGenes` method. But do we really need them?
If no, then we can remove them and with that decrease the dependency on JetBrains.